server/src/controller/authController.js

```diff
@@ -0,0 +1,42 @@
+import bcrypt from "bcrypt";
+import jwt from "jsonwebtoken";
+import User from "../models/user.js";
+
+export const register = async (req, res) => {
+  try {
+    const { firstName, lastName, email, password } = req.body;
+
+    const salt = await bcrypt.genSalt();
+    const passwordHash = await bcrypt.hash(password, salt);
+
+    const newUser = new User({
+      firstName,
+      lastName,
+      email,
+      password: passwordHash,
+    });
+    const savedUser = await newUser.save();
+
+    //dont know if sending back hashed password is still a security risk
```

Comment:
yes, i believe there is a bit of a security risk there. it's better to verify users at the backend first, then sign a token, and send email and token to the client instead of email and hashed password.